docs/design/features/readytorun-elf-support-roadmap.md

```diff
@@ -0,0 +1,166 @@
+  This is junior level task estimated at 3 man-weeks including creating a validation
+  test that should use the designed API to rewrite an arbitrary ELF executable and
+  verify that existing tools like readelf properly parse the file.
+  
+  It doesn't need to have 100% fidelity but we need to verify it's not distorted
+  and that the main components like targeting OS / architecture, pointers to the
+  sections and symbols are kept intact. The transcript from the dumping tool
+  should be made part of the output from the regression test in order to make
+  potential regressions easy to diagnose.
+  
+  We should also try to figure out whether we might be able to implement the
+  section builder logic in a similar manner for both PE and ELF builder - this
+  would probably require creating a wrapper over the PE builder but it would be
+  still easier than having two completely different code paths for composing the
+  sections. We'll need to address minute differences like alignment considerations
+  as on Linux **file offset - RVA** deltas need to be page-aligned.
+
+* Virtualizing R2RDump header parsing to add ELF support. This should be
+  junior-level task estimated at 1 man-week of work. It should check for the
+  well-known signature magic values used by PE / ELF files to identify the format
+  of the input file (both for the "main" and reference input files) and use
+  proper logic to locate the R2R header, COR header / MetadatReader and the logic
+  to map RVA's to file offsets.
+
+* Virtualizing Crossgen2 SectionBuilder / R2RPEBuilder: this is senior / principal
+  work estimated at 3 man-weeks to refactor section building to be independent on
+  the choice of PE / ELF format and use the newly created ELF API to [re]construct
+  the necessary headers and create a fully functional ELF executable.
+
+  At this point this will probably still be just a dynamic library, not a
+  full-fledged EXE. The full-fledged EXE should come in parallel with the
+  single-exe effort as it basically involves emitting some host bootstrapper into
+  the executable. Without the bootstrapper we can happily continue doing more or
+  less what we're doing today just with slightly different logic for locating
+  the headers.
+
+* Implementing CoreCLR runtime counterpart of PEFile and PEImage for ELF and
+  adding Linux-specific format detection to choose between PE and ELF as it's
+  expected that for a certain period of time we'll need to support both formats
+  on Linux. This is a senior / principal-level task estimated at 4 man-weeks
+  of work.
+  
+  It will require initial investigation as to what native code we can reuse
+  from the existing Linux libraries. We should consider some level of
+  virtualization to cater for common constructs. The new objects should be
+  Linux-specific and should use the Linux-native functions for manipulating
+  executables like **dlopen** / **dlsym**.
```

Comment:
Nit: The markdown convention is to quote identifiers using backticks

(multiple places)

Reply:
Fixed even though this is actually the only place I found - are you aware of others?